src/helpers/git-sourcemaps-repository-data.ts

```diff
@@ -2,7 +2,7 @@ import fs from 'fs'
 import * as simpleGit from 'simple-git'
```

Comment:
The `TrackedFilesMatcher` class does not seem tied to git particularly, could it be moved to a `sourcemap.ts` helpers file (to be shared amongst the two sourcemaps commands)?
That way this file could be merged with `git-span-metadata.ts` and be only `git.ts` with only git specific functions.

Reply:
@lraucy what is your opinion on this?  
@m-rousse what would be the reason to merge both files?

I believe merging all the git-related stuff in one file is decreasing the [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)) of the code, which is not a good thing: this file is going to be harder to maintain, if we couple both the span metadata and sourcemaps repository data it's likely to create bugs and since both are used by separate team it's going to take a lot of resources to solve this.
We should split files based on their responsibilities, not on the tools they used. It's not likely that we'd have to make a change to both the span metadata and sourcemaps data at the same time.

Reply:
Goal is to have a simpler grouping of functions, having `functions related to git` and `sourcemap helpers` would make re-using git related functions in other commands easier than having `function to detect git context` (the "span metadata" one) and `functions to get git metadata formatted for sourcemap`.
The abstraction I suggest is data oriented and not command oriented to make it easier to navigate the code when implementing commands that will rely on these helpers.

Reply:
As discussed, I've created a `git` folder with 3 files:
- `get-git-data` that contains git commands 
- `format-git-sourcemaps-data` and `format-git-span-data` that contains the logic to format the data for these cases

Upon reflexion, I am not sure an `index` file is needed, with auto-import it is still easy to import the files and the module is still small